### PR TITLE
Add `NextLockId` Query for lockup module

### PR DESCRIPTION
**Pull request by mattverse**
## What is the purpose of the change
Adds `NextLockID` query to the lock up module.
This PR adds the following:
- basic query logic for `NextLockID`
- introduces cli for `NextLockID` query
- adds query to stargate whitelist

## Brief Changelog
- Add `NextLockId` Query for lockup module
## Testing and Verifying
This PR adds tests.

CLI command is not tested.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: proto/osmosis/protorev/v1beta1/query.proto [hunk 1]
```diff
@@ -245,8 +245,10 @@ message QueryGetProtoRevPoolWeightsRequest {}
 // Query/GetProtoRevPoolWeights RPC method.
 message QueryGetProtoRevPoolWeightsResponse {
   // pool_weights is a list of all of the pool weights
-  PoolWeights pool_weights = 1
-      [ (gogoproto.moretags) = "yaml:\"pool_weights\"", (gogoproto.nullable) = false ];
+  PoolWeights pool_weights = 1 [
+    (gogoproto.moretags) = "yaml:\"pool_weights\"",
```
Comment (mattverse) [inline proto/osmosis/protorev/v1beta1/query.proto:249]: changes in linting due to proto dependency, can be ignored

File: x/lockup/client/cli/query.go [hunk 2]
```diff
@@ -185,6 +186,14 @@ func GetCmdLockedByID() *cobra.Command {
 	return osmocli.BuildQueryCli[*types.LockedRequest](&q, types.NewQueryClient)
 }
 
+// GetCmdSyntheticLockupsByLockupID returns synthetic lockups by lockup id.
```
Comment (mattverse) [inline x/lockup/client/cli/query.go:189]:
```suggestion
// GetCmdNextLockID returns next lock id to be created.
```

**Review by ValarDragon — APPROVED** [review]
Perfect, thank you!

**Comment by ValarDragon** [comment]
Need to update the hardcoded 63 in ICQ params length tho https://github.com/osmosis-labs/osmosis/blob/main/app/upgrades/v15/upgrade_test.go#L204

**Review by p0mvn — APPROVED** [review]

**Pull request merged**